### PR TITLE
Load framework defaults for Rails 7.2

### PR DESCRIPTION
**Pull request by martinemde**
Removes the yjit initializer. YJIT is now enabled by default.

### CODE REVIEW

**Review by segiddins — APPROVED** [review]

File: config/initializers/new_framework_defaults_7_2.rb [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-
-###
-# Controls whether the PostgresqlAdapter should decode dates automatically with manual queries.
-#
-# Example:
-#   ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.select_value("select '2024-01-01'::date") #=> Date
-#
-# This query used to return a `String`.
-#++
-# Rails.application.config.active_record.postgresql_adapter_decode_dates = true
```
Comment (martinemde) [inline config/initializers/new_framework_defaults_7_2.rb:64]: We have no date fields.

File: config/initializers/new_framework_defaults_7_2.rb [hunk 2]
```diff
@@ -1,70 +0,0 @@
...
-
-###
-# Enables YJIT as of Ruby 3.3, to bring sizeable performance improvements. If you are
-# deploying to a memory constrained environment you may want to set this to `false`.
-#++
-# Rails.application.config.yjit = true
```
Comment (martinemde) [inline config/initializers/new_framework_defaults_7_2.rb:70]: We already enable yjit ourselves.

**Review by martinemde — COMMENTED** [review]
Just notes for future reference on whether or not these are likely to cause any problems.

**Comment by martinemde** [comment]
The tests that fail here _do not_ fail on the rails 8 branch. This is caused by a framework default change, almost certainly the job transaction/commit behavior. Something in rails 8 may have fixed the behavior in tests so that this no longer occurs.

**Comment by martinemde** [comment]
Given that these tests fail and rails 8 tests, with rails 8 defaults pass, I'm guessing something about how we run these failing tests was fixed in rails 8. I do not think it's worth it to try to fix this test now when I know my fix is unnecessary to get us to rails 8.

**Pull request closed without merging**